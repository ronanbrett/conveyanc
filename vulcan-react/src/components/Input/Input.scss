@import "src/styles/vars";

.Input {
  padding: var(--spacing-m) var(--spacing-m);
  border: 2px solid var(--border-color);

  font-size: 1em;

  &:focus {
    outline: none;
    border-color: var(--highlight-color);
  }

  &-large {
    @include type-header-m;
  }

  &.field--invalid {
    border: 2px solid var(--warn-color);
  }
}
