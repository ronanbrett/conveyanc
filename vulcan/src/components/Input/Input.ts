import { useModelWrapper } from '@/core/utils/forms.utils';
import { defineComponent, onMounted, ref } from 'vue';

const Input = defineComponent({
  props: {
    iconAfter: String,
    loading: Boolean,
    color: String,
    modelValue: String,
    autocomplete: {
      type: String,
      default: 'off',
    },
    label: {
      type: String,
      default: '',
    },
    placeholder: {
      type: String,
      default: '',
    },
    value: {
      type: String,
      default: '',
    },
  },
  setup(props, { emit }) {
    const val = useModelWrapper(props, emit);
    const innerValue = `${val.value}`;

    const onBlur = () => {
      console.log('blur!');
      emit('blur');
    };

    return {
      val,
      innerValue,
      onBlur,
    };
  },
});

export default Input;
