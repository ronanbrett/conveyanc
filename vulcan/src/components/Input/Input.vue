<template>
  <input
    type="text"
    :value="val"
    @blur="onBlur"
    @input="val = $event.target.value"
    :autocomplete="autocomplete"
    :placeholder="placeholder"
  />
</template>

<script src="./Input.ts" lang="ts"></script>

<!-- Add "scoped" attribute to limit SCSS to this component only -->
<style scoped lang="scss">
@import './src/styles/vars';

input {
  padding: var(--spacing-m) var(--spacing-m);
  border-radius: var(--border-radius);
  border: 1px solid var(--border-color);

  &:focus {
    outline: none;
    border: 2px solid var(--highlight-color);
  }
}
</style>
