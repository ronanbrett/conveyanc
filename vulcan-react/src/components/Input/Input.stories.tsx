import React from "react";

import Input from "./Input";

export default {
  title: "Components/Input",
  decorators: [],
  component: Input,
};

export const Default = (props) => <Input {...props}></Input>;

export const WithFormikField = (props) => {
  return <Input {...props}></Input>;
};
