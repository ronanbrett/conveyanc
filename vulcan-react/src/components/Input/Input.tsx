import { addFieldValidationClasses } from "@core/utils/field.utils";
import {
  FieldConfig,
  FieldInputProps,
  FieldMetaProps,
  FieldProps,
  FormikFormProps,
  useField,
} from "formik";
import React, { FC } from "react";

import "./Input.scss";

type InputTypes =
  | "email"
  | "month"
  | "number"
  | "password"
  | "search"
  | "tel"
  | "text"
  | "time"
  | "url"
  | "week";

type InputProps<T> = {
  type?: InputTypes;
  autocomplete?: string;
  children?: any;
  size?: "large" | "small";
  id?: string;
  placeholder?: string;
  value?: any;
};

const Input = <T extends any>({
  type,
  autocomplete,
  size,
  ...props
}: InputProps<T> & FieldConfig) => {
  const [field, meta, helpers] = useField(props);

  return (
    <input
      className={`Input Input-${size} ${addFieldValidationClasses(meta)}`}
      type={type}
      autoComplete={autocomplete}
      {...field}
      {...props}
    />
  );
};

Input.defaultProps = {
  type: "text",
  autocomplete: "off",
  size: "large",
};

export default Input;
