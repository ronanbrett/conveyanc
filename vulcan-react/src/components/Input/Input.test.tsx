import React from 'react';
import { render } from '@testing-library/react';
import { create } from 'react-test-renderer';
import Input from './Input';
import { Field, Form, Formik } from 'formik';




describe('Input', () => {
   it('renders without error', () => {
      const component = create(
         <Formik
            initialValues={{ propertyType: "TestC" }}
            onSubmit={(x) => console.log(x)}
         >
            <Form>
               <label htmlFor="propertyType">Property Type</label>
               <Input id="name" name="name" placeholder="John" />
            </Form>
         </Formik>
      )

      expect(component).toMatchSnapshot();
   });
});