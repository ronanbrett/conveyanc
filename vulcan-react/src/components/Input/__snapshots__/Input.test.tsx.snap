// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders without error 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <label
    htmlFor="propertyType"
  >
    Property Type
  </label>
  <input
    autoComplete="off"
    className="Input Input-large field--valid"
    id="name"
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="John"
    type="text"
  />
</form>
`;
